### ISSUE TITLE
Passing `frozen_default=True` to `dataclass_transform()` ignores value passed to `frozen`

### ISSUE THREAD

**Issue by bryanforbes** [microsoft/pyright#4444]
Note: if you are reporting a wrong signature of a function or a class in the standard library, then the typeshed tracker is better suited for this report: https://github.com/python/typeshed/issues.

**Describe the bug**
Given the following:

```python
from __future__ import annotations

from collections.abc import Callable
from dataclasses import dataclass, field
from functools import partial
from typing import TYPE_CHECKING, TypeVar, overload
from typing_extensions import dataclass_transform

_T = TypeVar('_T')

if TYPE_CHECKING:

    @overload
    @dataclass_transform(frozen_default=True, field_specifiers=(field,))
    def frozen(__cls: type[_T]) -> type[_T]:
        ...

    @overload
    @dataclass_transform(frozen_default=True, field_specifiers=(field,))
    def frozen(__cls: None) -> Callable[[type[_T]], type[_T]]:
        ...

    @overload
    @dataclass_transform(frozen_default=True, field_specifiers=(field,))
    def frozen(
        *,
        init: bool = ...,
        repr: bool = ...,
        eq: bool = ...,
        order: bool = ...,
        unsafe_hash: bool = ...,
        frozen: bool = ...,
        match_args: bool = ...,
        kw_only: bool = ...,
        slots: bool = ...,
    ) -> Callable[[type[_T]], type[_T]]:
        ...

    def frozen(*args: Any, **kwargs: Any) -> Any:
        ...

else:
    frozen = partial(dataclass, frozen=True)


@frozen
class Foo:
    id: int


@frozen(frozen=False)
class Bar:
    id: int


f = Foo(1)
b = Bar(1)
b.id = 2  # Error: Bar is frozen (incorrect)
f.id = 2  # Error: Foo is frozen (correct)
```

Line 58 is incorrectly marked as an error.

**Expected behavior**
Line 58 should not be marked as an error.

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line

Which version?
1.1.288

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.289, which I just published. It will also be included in a future release of pylance.
